### PR TITLE
Update docker compose docs re private fork

### PR DESCRIPTION
**Pull request by JeanPaulSG**
## Test plan

Technical review by CE in PR, content editing by Product Education team members.

### CODE REVIEW

**Comment by JeanPaulSG** [comment]
With reference to the following change: https://github.com/sourcegraph/sourcegraph/issues/30572 
and the following discussion: https://sourcegraph.slack.com/archives/C02E4HE42BX/p1652872129430089 
@Rhia2

**Comment by JeanPaulSG** [comment]
CLA signed.

File: doc/admin/deploy/docker-compose/index.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ The sections below cover the process to [create a fork](#create-a-fork), [clone]
 
 [Create a fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) of the [sourcegraph/deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/) [reference repository](#reference-repository).
 
-> WARNING: Set your fork to **private** if you plan to store secrets (SSL certificates, external Postgres credentials, etc.) within the repository.
+> WARNING: If you plan to store secrets (SSL certificates, external Postgres credentials, etc.) within the repository you should duplicate the reference repository and make your copy private. To create a private copy you create an empty private repository in GitHub, bare clone the reference repository, mirror push the bare clone to your new private repository, and remove your local bare clone.
```
Comment (caugustus-sourcegraph) [inline doc/admin/deploy/docker-compose/index.md:40]: What do you think about writing out these commands explicitly? Translating that warning into exact commands is one extra hurdle for a customer to get started.

Another approach might be to link to the [Github instructions](https://docs.github.com/en/repositories/creating-and-managing-repositories/duplicating-a-repository#mirroring-a-repository) for mirroring, but that still requires the customer to translate the commands into a Sourcegraph-specific context, and it interrupts the flow. by sending them to another page.

Comment (JeanPaulSG) [reply doc/admin/deploy/docker-compose/index.md]: I added the terminal commands for creating a private duplicate. I also pulled the instruction out of the warning and made it clear that it was an alternative or parallel path to creating a public fork.

**Review by caugustus-sourcegraph — COMMENTED** [review]
Looks really good! One small change request to avoid assuming a dependency on Github

File: doc/admin/deploy/docker-compose/index.md [hunk 2]
```diff
@@ -27,17 +27,38 @@ Our Docker Compose support also has the following requirements:
...
+
+#### Create a fork or private duplicate
 
-The sections below cover the process to [create a fork](#create-a-fork), [clone](#clone-your-fork), [configure a release branch](#configure-release-branch), and [make YAML customizations](#make-yaml-customizations). Once completed you'll be ready to [run](#run-sourcegraph) Sourcegraph.
+[Create a public fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) of the [sourcegraph/deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/) [reference repository](#reference-repository).
 
-#### Create a fork
+Alternatively, create a [private duplicate](https://docs.github.com/en/repositories/creating-and-managing-repositories/duplicating-a-repository#mirroring-a-repository) of the [sourcegraph/deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/) [reference repository](#reference-repository) as follows:
+
+Create an [empty private repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-new-repository) in GitHub, for example <you/private-repository>, then bare clone the reference repository.
```
[the remark this answers is not in the document]
Comment (JeanPaulSG) [reply doc/admin/deploy/docker-compose/index.md]: Understood. I moved the mention of GitHub to the example part of the sentence. So it isn't a requirement, but is used in the example code.

**Review by caugustus-sourcegraph — APPROVED** [review]
🎉 Thank you for updating this!

**Pull request merged**